segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -605,6 +607,17 @@ void commit(DataFrameBuilder.CommitArgs commitArgs) {
             }
         }
 
+        private void queueMemoryCommit(List<List<CompletableOperation>> toAck) {
+            try {
+                toAck.forEach(OperationProcessor.this.commitQueue::add);
+            } catch (Exception ex) {
+                // If we are unable to queue up these operations for whatever reason, we must unregister them from the
+                // CacheUtilizationProvider, otherwise we will be "leaking" bytes (i.e., overcounting).
+                toAck.forEach(l -> l.forEach(this::notifyOperationCommitted));
```

Comment:
If you look at what that method does, it decrements a byte counter. If we failed the operation we need to subtract its effect from the counter that we use for cache throttling.

Reply:
Ok, I checked the method and it only adjust pending bytes, so to confirm, you are saying that the adjustment needs to be done independent of whether we can add the operation to the commit queue at this point, yes? There is no other important side-effect, correct? Please confirm that to me.

Reply:
When we pick operations from the OP queue, we need to check how many bytes they would eventually add to the cache. We use this aggregated amount to calculate how much to throttle (otherwise we risk accepting more data than we can eventually fit into the cache - in cases of ingestion spikes).

We need to unaccount for these bytes when:
1. We add those operations to the cache (commit success)
2. We reject those operations
3. We fail the operations (because of bad Tier 1 writes, because of OP shutting down or other failure reasons).

1 and 2 are already accounted for in `processOperations`. For 3, we covered all the paths of failure except one. The one that was missing was if those operations have been acked from Tier 1 but were in the process of being added to the commit queue; the commit queue has just been closed so we fail them there. This change here plugs this hole.